### DISCUSSION TITLE
Getting Parse error

### DISCUSSION THREAD

**Discussion by mahesh99599** [KaTeX/KaTeX#2791] [General] [answered]
Hi, I'm not able to parse Katex "$\Ia = \sum{i=1}^\infty\ointM r{Q}^{2}\theta \,dA$ or$ \begin{Bmatrix} F = 12 N \\cr\tau = 26 N.m\end{Bmatrix} $" and it's giving "KaTeX parse error: Expected 'EOF', got '\' at position 8: \Ia = \̲\̲sum{i=1}^\inf… or KaTeX parse error: Expected 'EOF', got '\' at position 1: \̲\̲begin{Bmatrix} …", Can anyone help me in this, Thanks in advance

**Comment by mbourne** [comment, marked as answer]
This may not be exactly what you want, but this works:
Ia = \sum_{i=1}^\infty\oint M r{Q}^{2}\theta ,dA\ \text{or}\ \begin{Bmatrix} F = 12 N \cr\tau = 26 N.m\end{Bmatrix}

![image](https://user-images.githubusercontent.com/7541940/108016344-43f73800-704d-11eb-9a32-9486d0e99504.png)

Hope it helps.
Regards
Murray

**Reply by mahesh99599** [reply]
Thanks for your quick response @mbourne. This is what exactly I'm looking for, Thank you soo much. And do we have any online editor for checking Katex format ?

**Reply by mbourne** [reply]
The KaTeX home page has an editor, and is where I took the screen shot: https://katex.org/

**Reply by mahesh99599** [reply]
Yea, Thank you @mbourne

**Reply by mahesh99599** [reply]
Hi, As per my requirement I have to import the Katex syntax with CSV file. But due to commas in between the columns are escaping. My syntax like "Ia = \sum_{i=1}^\infty\oint M r{Q}^{2}**\theta ,dA**\ \text{or}\ \begin{Bmatrix} F = 12 N \cr\tau = 26 N.m\end{Bmatrix}" highlighted. Is there any delimiter to replace comma ?

**Reply by edemaine** [reply]
If you wrap the CSV entry in double quotes, it can include a comma. See e.g. [this post](https://stackoverflow.com/questions/4617935/is-there-a-way-to-include-commas-in-csv-columns-without-breaking-the-formatting).

**Reply by mahesh99599** [reply]
Hi @edemaine Thanks for your response, If you don't mind can u share me an example with Katex if you have any ? I'm not able to display with double quotes

**Reply by edemaine** [reply]
No, I don't have any examples; I've never used KaTeX with CSV. I'm just explaining how the CSV spec works, assuming you're using a proper CSV parser. If it's not working, perhaps you could use TSV instead, which is simpler.

**Reply by mbourne** [reply]
I interpreted the question differently. I took it to mean @mahesh99599 is getting unwanted commas when importing math expression segments from a CSV. 

If so, he would need to do a replace

So for example, to replace the 2 unwanted commas in this expression, do:

    var txt = "\sum_{i=1}^\infty\oint M, r{Q}^{2}\theta ,dA"
    var fixedTxt = txt.replace(/,/g, '');  

Then feed `fixedTxt` to KaTeX inside $ signs.

**Reply by mahesh99599** [reply]
Hi @mbourne with this approach we can achieve, but when we have lot we can't do this by replacing each column r row. for normal case we will give double quotes instead of comma, I'm looking for any alternate like double quotes for Katex

**Reply by mbourne** [reply]
@mahesh99599 I'm not really clear about the problem you're facing.

If you gave us a sample (a few lines) of the CSV file you are importing from, and what you hope to happen, perhaps we can help.

**Reply by mahesh99599** [reply]
@mbourne I attached the csv file, In that 5th and 6th row due to comma in between katex formula, the columns are separating. As per my requirement all formulas should be in classDefinition column but due to comma it is moving to next column


[Katex import.xlsx](https://github.com/KaTeX/KaTeX/files/6026524/Katex.import.xlsx)

**Reply by mbourne** [reply]
Ah, now I see the issue.

I agree with @edemaine : Best is to output your file with tab separators, not commas. That's what I did to achieve the attached file.
[Katex.import.txt](https://github.com/KaTeX/KaTeX/files/6026621/Katex.import.txt)

**Reply by mahesh99599** [reply]
Okay, and thank you @mbourne & @edemaine
